Lift update binding logic from DuckTableEntry to TableCatalogEntry

bleskes opened the pull request:
Binding the update operator requires some careful handling of indices. All of that currently happens in the DuckTableEntry class but it seems there's nothing really DuckTable specific in it. We can lift it to the base class TableCatalogEntry, so the logic is available for custom catalog implementation.

pdet committed:
Start to expose the arrow buffer size option

carlopi committed:
Add serialization for BoundOrderModifier

carlopi committed:
Serialize ORDER BY in BoundAggregateExpression

carlopi committed:
Serialize ORDER BY in LogicalDistinct

pdet committed:
More on adding arrow_large_buffer_size setting

pdet committed:
Ensure the option works on python

pdet committed:
R

pdet committed:
Bring timezone to arrowoptions

pdet committed:
Test reset fix

pdet committed:
Tidy fixes

pdet committed:
Thijs' feeback

pdet committed:
Remove IDE include

Tishj committed:
apply feedback, partially

carlopi committed:
Implement serialization for BoundDefaultExpression

hannes committed:
add duckdb:::rel_from_table_function for R

hannes committed:
basic test

carlopi committed:
List aggregates serialization

hannes committed:
mc kirill's suggestions

pdet committed:
Update arrow oot extension

carlopi committed:
BindLambdaFunction: Remove unneded changes to function.arguments

Together with @taniabogatsch

hannes committed:
trying to disable hive partitioning on R relational test case to not trigger UBSAN on RE2

carlopi committed:
First/last/(any): aggregate: Fix binder (triggered via serialization)

carlopi committed:
Fix uninitialized value in GetDecimalProperties [msan]

pdet committed:
I changed the wrong tag

hannes committed:
removed offending test case entirely

Commit:
Enhance the generation of random floating point

Tishj committed:
only execute in Close if it was never executed before

Mytherin committed:
Reservoir sample - when appending a chunk to two different reservoirs we need to explicitly slice it

Mytherin committed:
Correctly check for nullpointer in SQLite shell function

Mytherin committed:
Correctly deal with extreme numbers in round precision

Mytherin committed:
Fix negation underflow in list extract

Mytherin committed:
Correctly throw exception on empty set returned by COLUMNS() expression in WHERE clause

Mytherin committed:
Correctly check for NULL in database

Mytherin committed:
Array slice underflow

Mytherin committed:
Correctly check if time is valid in make_time

Commit:
Fix the unit test that relies on a particular seed value

Mytherin committed:
Qualify binder - correctly check that all columns appear in the GROUP BY list or in aggregates

Mytherin committed:
OssFuzz fixes

adavis444 committed:
Build Python wheels for aarch64 using QEMU

Mytherin committed:
Missing rowsort in test

Tishj committed:
achieve the same behavior, but return None from the __exit__ method instead, fixes #7816

Mytherin committed:
Disable attaching on-disk DuckDB databases if external access is disabled

Mytherin committed:
Fix #7711 - disallow detaching the currently USEd database

Mytherin committed:
More coverage fixes

pdet committed:
Update arrow tag

carlopi committed:
Fix warning: unused debug variables in ART

carlopi committed:
Fix warning: unused variable in ICU

Mytherin committed:
More coverage

hannes committed:
fix for table in function binding

Mytherin committed:
Fix #7795 - provide explicit alias for unnamed subquery as unpivot parameter

Mytherin committed:
This succeeds

Mytherin committed:
Disable this test entirely for now

Commit:
grab mark join lock

Commit:
add test for parallel mark join

Mytherin committed:
JSON Contains - correctly handle constant NULL

pdet committed:
New wasm tag

carlopi committed:
Rework test_plan_serialization_bwc to roundtrip serialization -> deserialization

Generation test is now a roundtrip test, either on ignored temporary file OR on
file to be actually committed (if run with GEN_PLAN_STORAGE environment set, either
manually or via `python3 scripts/generate_plan_storage_version.py`

Mause committed:
fix(nodejs): another http state error

Maxxen committed:
bump spatial

taniabogatsch committed:
fix unnest(null) behavior, close #7391

Mytherin committed:
statement maybe

Mause committed:
test(nodejs): test for http state

Maxxen committed:
add spatial functions

stephaniewang526 committed:
feat: update parser exception handling for extensions

bleskes committed:
lift update bind

Maxxen committed:
adjust regex

carlopi committed:
Adapt to feedback from @Mytherin on TestCreatePath

Now we don't have a temp file hanging anymore

carlopi committed:
Function Deserialization rework

Deserialization signature to take PlanDeserializationState& instead of ClientContext&

Flogex committed:
Implement DatabaseManager::SetDefaultDatabase

The `DatabaseManager::default_database` is returned by `DatabaseManager::GetDefaultDatabase`
when the default catalog in the search path is invalid. This commit adds a method to change
`default_database`.
This makes it possible that new `Connection`s can use a database other than 'memory' as default.

carlopi committed:
Regression.yml to benchmark agaist target branch (or 'master')

GITHUB_BASE_REF will be equal to the target branch for pull requests, and
undefined, otherwise.
(docs: https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)

Given we use the bash syntax to use variable or default
(= to 'master'), this should make that pull requests against feature (or any
other branch) will be tested only that given branch.

Tishj committed:
avoid constant folding, and don't reduce the result to a constant vector

Tishj committed:
add keyword arg for 'side_effects'

Tishj committed:
tweaks

pdet committed:
This issue seems unrelated to this PR, but adding a lock to verify method of parallel reader

Flogex committed:
Use 'ClientContext::RunFunctionInTransaction' in 'SetDefaultDatabase'

pdet committed:
bump substrait tag

Mytherin commented:
Thanks for the PR! LGTM. This might require some tweaking in the SQLite extension as well. The single-file compilation (compiling with `DISABLE_UNITY=1`) is also failing in the CI - likely due to some missing includes. Could you have a look at that?

stephaniewang526 committed:
address feedback; better error handling

Mytherin committed:
CI fix

stephaniewang526 committed:
improve logic

stephaniewang526 committed:
fix capi test failure

stephaniewang526 committed:
remove un-used var

stephaniewang526 committed:
fix logic

adavis444 committed:
Skip build of musllinux aarch64 wheels

carlopi committed:
Bump up uncovered_files.csv

Mause committed:
fix: shared, not unique

not sure how this happened

bleskes committed:
missing includes

bleskes commented:
@Mytherin the compilation issue should be fixed now. Any tips on how to the SQLite extension? I'm not sure what the process is because in another repo

Mytherin commented:
Generally we send a PR and merge it into the other repo first, and then update extensions.csv with the new git hash as part of the PR

carlopi committed:
flatten.test: Fix by ordering both table and lists

douenergy committed:
fix pivotref comments

Mytherin committed:
uncovered files

pdet committed:
Update uncovered files for CodeCov CI

bleskes committed:
tweak

bleskes commented:
@Mytherin I've put up a pr to sql lite, but I'm clearly not following the right process to validate it. Can you point me at where I can read the steps?

pdet committed:
somehow i renamed this whole thing by accident

stephaniewang526 committed:
more feedback

hannes committed:
fixing python map

Mytherin commented:
The SQLite scanner still uses the old extension mechanism which is not very well documented. You can update `extensions.csv` in this PR (and run the format fix) in order to move the SQLite version forward.

Flogex committed:
Removed call to 'RunFunctionInTransaction' in 'DatabaseManager::SetDefaultDatabase'

Most functions that call 'SetDefaultDatabase' will already have an active transaction,
hence it is not necessary to create a new one.

stephaniewang526 committed:
move guards

bleskes committed:
feedback

bleskes commented:
@Mytherin done. Thanks.

stephaniewang526 committed:
more codecov shenanigans

carlopi committed:
__builtin_clzll(0) is UB: restructure bit_utils [duckfuzzer]

Fixes https://github.com/duckdb/duckdb-fuzzer/issues/285

carlopi committed:
top_level_pivot_syntax.test: missing rowsort

Mytherin committed:
Fix #7902 - add support for blob to approx count distinct

Mytherin committed:
Update database_manager.cpp

Mytherin committed:
Test fixes + codecov ignores

bleskes commented:
@Mytherin lmk if the thread santinzer issue looks to you like it was caused by this pr.

Mytherin committed:
Grab update lock in GetStorageInfo

bleskes committed:
feebdack

bleskes committed:
Merge remote-tracking branch 'duckdb_labs/master' into update_static_bind_methods

carlopi commented on src/include/duckdb/catalog/catalog_entry/table_catalog_entry.hpp:
```diff
@@ -37,7 +37,7 @@ class TableFunction;
 
 class TableColumnInfo;
-class TableIndexInfo;
+class ColumnSegmentInfo;
```
```suggestion
struct ColumnSegmentInfo;
```

carlopi commented on src/include/duckdb/storage/table/row_group.hpp:
```diff
@@ -31,7 +31,7 @@ class RowGroupCollection;
 class UpdateSegment;
 class TableStatistics;
-class TableStorageInfo;
+class ColumnSegmentInfo;
```
```suggestion
struct ColumnSegmentInfo;
```

carlopi commented on src/include/duckdb/storage/table/row_group_collection.hpp:
```diff
@@ -26,6 +26,7 @@ struct TableAppendState;
 class DuckTransaction;
 class BoundConstraint;
 class RowGroupSegmentTree;
+class ColumnSegmentInfo;
```
```suggestion
struct ColumnSegmentInfo;
```

carlopi reviewed (commented):
+ I would add line:
```
catalog/catalog_entry/table_catalog_entry.cpp 3
```
somewhere in .github/config/uncovered_lines.csv.

Thanks!

Mytherin committed:
Apply suggestions from code review

Co-authored-by: Carlo Piovesan <piovesan.carlo@gmail.com>

Mytherin committed:
Add uncovered file

bleskes committed:
one more struct
